[lldb/crashlog] Add `--no-parallel-image-loading` hidden flag

medismailben opened the pull request:
This patch adds the `--no-parallel-image-loading` to the crashlog command. By default, image loading will happen in parallel in the crashlog script however, sometimes, when running tests or debugging the crashlog script itself, it's better to load the images sequentially.

As its name suggests, this flag will disable the default image loading behaviour to load all the images sequencially in the main thread.

JDevlieghere commented on lldb/examples/python/crashlog.py:
```diff
@@ -555,30 +555,44 @@ def load_images(self, options, loaded_images=None):
+                return image, image.add_module(target, obj_dir)
+
+            if options.parallel_image_loading:
+                with concurrent.futures.ThreadPoolExecutor() as executor:
```
Instead of duplicating the code, I would limit the number of workers in the ThreadPool to one.

medismailben replied:
Good point!

JDevlieghere commented on lldb/examples/python/crashlog.py:
```diff
@@ -1720,6 +1735,13 @@ def CreateSymbolicateCrashLogOptions(
         default=False,
     )
+    arg_parser.add_argument(
+        "--no-parallel-image-loading",
+        dest="parallel_image_loading",
```
The name of the flag and the variable seem to be opposites. Does argparse automatically change the value because the flag contains `-no`.

JDevlieghere commented on lldb/examples/python/crashlog.py:
```diff
@@ -1720,6 +1735,13 @@ def CreateSymbolicateCrashLogOptions(
+        dest="parallel_image_loading",
+        action="store_false",
+        help=argparse.SUPPRESS,
+        default=True,
```
Does this mean parallel loading is on or off by default? The negation confuses me.

medismailben committed:
[lldb/crashlog] Add `--no-parallel-image-loading` hidden flag

This patch adds the `--no-parallel-image-loading` to the crashlog
command. By default, image loading will happen in parallel in the
crashlog script however, sometimes, when running tests or debugging the
crashlog script itself, it's better to load the images sequentially.

As its name suggests, this flag will disable the default image loading
behaviour to load all the images sequencially in the main thread.

Signed-off-by: Med Ismail Bennani <ismail@bennani.ma>
